sdk/servicebus/azure-servicebus/azure/servicebus/exceptions.py

```diff
@@ -228,108 +193,245 @@ class ServiceBusError(AzureError):
+        try:
+            message = message.decode('UTF-8')
+        except AttributeError:
+            pass
+
+        if self._condition:
+            try:
+                condition = self._condition.decode('UTF-8')
+            except AttributeError:
+                condition = self._condition
+            message = message + "\nError condition: {}".format(str(condition))
+        if self._status_code is not None:
+            message = message + "\nStatus Code: {}".format(str(self._status_code))
+        super(ServiceBusError, self).__init__(message, *kwargs, **kwargs)
 
 
 class ServiceBusConnectionError(ServiceBusError):
     """An error occurred in the connection."""
-
-
-class ServiceBusAuthorizationError(ServiceBusError):
-    """An error occurred when authorizing the connection."""
+    def __init__(self, **kwargs):
+        message = kwargs.pop("message", None) or "An error occurred in the connection."
+        super(ServiceBusConnectionError, self).__init__(
+            message,
+            retryable=True,
+            shutdown_handler=True,
+            **kwargs
+        )
 
 
 class ServiceBusAuthenticationError(ServiceBusError):
     """An error occurred when authenticate the connection."""
+    def __init__(self, **kwargs):
+        message = kwargs.pop("message", None) or "An error occurred when authenticate the connection."
+        super(ServiceBusAuthenticationError, self).__init__(
+            message,
+            retryable=True,
+            shutdown_handler=True,
+            **kwargs
+        )
 
 
-class NoActiveSession(ServiceBusError):
-    """No active Sessions are available to receive from."""
+class UnauthorizedAccessError(ServiceBusError):
+    """An error occurred when authorizing the connection."""
+    def __init__(self, **kwargs):
+        message = kwargs.pop("message", None) or "An error occurred when authorizing the connection."
+        super(UnauthorizedAccessError, self).__init__(
+            message,
+            retryable=True,
+            shutdown_handler=True,
+            **kwargs
+        )
 
 
 class OperationTimeoutError(ServiceBusError):
     """Operation timed out."""
+    def __init__(self, **kwargs):
+        message = kwargs.pop("message", None) or "Operation timed out."
+        super(OperationTimeoutError, self).__init__(
+            message,
+            retryable=True,
+            shutdown_handler=False,
+            **kwargs
+        )
 
 
-class ServiceBusMessageError(ServiceBusError):
-    """An error occurred when an operation on a message failed because the message is in an incorrect state."""
-
-
-class MessageContentTooLarge(ServiceBusMessageError, ValueError):
+class MessageSizeExceededError(ServiceBusError, ValueError):
     """Message content is larger than the service bus frame size."""
+    def __init__(self, **kwargs):
+        message = kwargs.pop("message", None) or "Message content is larger than the service bus frame size."
```

Comment:
Can we give the value here as part of the message?

Reply:
This is just a default message, we do the "provide from higher in the callstack" thing.